lib/posix/clock.c

```diff
@@ -15,21 +15,18 @@
  */
 int clock_gettime(clockid_t clock_id, struct timespec *ts)
 {
-	s64_t elapsed_msecs, elapsed_secs;
-	s64_t elapsed_nsec, elapsed_cycles;
+	s64_t elapsed_secs, elapsed_nsec;
+	u64_t elapsed_msecs;
 
 	if (clock_id != CLOCK_MONOTONIC) {
 		errno = EINVAL;
 		return -1;
 	}
 
 	elapsed_msecs = k_uptime_get();
-	elapsed_secs = elapsed_msecs / MSEC_PER_SEC;
-
-	elapsed_cycles = (s64_t) (k_cycle_get_32() %
-				  sys_clock_hw_cycles_per_sec);
-	elapsed_nsec = (s64_t) ((elapsed_cycles * NSEC_PER_SEC) /
-				sys_clock_hw_cycles_per_sec);
+	elapsed_secs = (s64_t) (elapsed_msecs / MSEC_PER_SEC);
+	elapsed_nsec = (s64_t) ((elapsed_msecs % MSEC_PER_SEC) *
+					USEC_PER_MSEC * NSEC_PER_USEC);
```

Comment:
Seems like it's still to complicated.  If you're going to do this in 64 bit math, don't bother finessing the components, you don't need to:

    u64_t ms = k_uptime_get();
    ts->tv_sec = (s32_t)(ms / 1000);
    ts->tv_nsec = (s32_t)((ms % 1000)*1000);

Also, IMHO it's perfectly fine to assume that the SI standard for unit naming is going to adhere in perpetuity and that humanity will never change the meaning of "micro" or "nano" such that the code needs to adapts.  Those x_PER_y constants seem sorta silly to me.

Reply:
will get rid of the local variables and those constants are from Zephyr include headers so will stick with them.